### DISCUSSION TITLE
No assurance on scheduled jobs?

### DISCUSSION THREAD

**Discussion by kingthorin** [community/community#27130] [Actions]
<p><sub>This is a <a href="https://github.community/t/scheduled-jobs-skipped/133172">re-post</a>, as I just realized I started the thread in the wrong forum. (And, don’t see any functionality to move it.)</sub></p>
<p>Is there no assurance on execution of scheduled jobs? I mean if they aren’t executed with in <em>n</em> time (window) of their scheduled time are they abandon? Could there have been some sort of GitHub Actions resource contention issue that caused them not to run/fire and then the window was passed?</p>
<p>Was there an infrastructure issue a few days ago that prevented scheduled jobs from running?</p>
<p>I’m asking because I don’t see any evidence of this scheduled job being run on the 18th.</p>
<p>Workflow: <a href="https://github.com/zapbot/zap-mgmt-scripts/blob/742d9a8c1170634ff5d135a38848f69f129a69a7/.github/workflows/record-stats.yml" rel="noopener nofollow ugc">https://github.com/zapbot/zap-mgmt-scripts/blob/742d9a8c1170634ff5d135a38848f69f129a69a7/.github/workflows/record-stats.yml</a>, as you can see it hasn’t been changed in 22 days. The schedule is set for midnight:</p>
<pre><code class="lang-auto">on:
  schedule:
    - cron:  '0 0 * * *'
</code></pre>
<p>But there’s no “3 days ago” run at all:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/39b7d8552409758e4be8fc3736251367db112c50.png" data-download-href="/uploads/short-url/8eB6k9sB1onsm6bDlEYTT9GrogE.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/7570458/181101992-9738c24c-e1f5-486f-82d1-db047c72e963.png" alt="image" data-base62-sha1="8eB6k9sB1onsm6bDlEYTT9GrogE" width="626" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/39b7d8552409758e4be8fc3736251367db112c50_2_626x500.png, https://aws1.discourse-cdn.com/github/original/2X/3/39b7d8552409758e4be8fc3736251367db112c50.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/3/39b7d8552409758e4be8fc3736251367db112c50.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/39b7d8552409758e4be8fc3736251367db112c50_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">934×745 35.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/kingthorin">@kingthorin</a> ,</p>
<p>Many times, there is a delay when triggering the scheduled workflow.<br>
Generally the delay time is about 3 to 10 minutes. Sometimes it may be more time, even dozens of minutes or more than one hour. However, if the delay time is too long, the scheduled workflow may be not triggered at that day.</p>
<p>In your case, you can try to use the “<a href="https://docs.github.com/en/rest/reference/actions#get-a-workflow-run" rel="noopener nofollow ugc"><strong>Get a workflow run</strong></a>” API to get the details of runs <strong><span class="hashtag">#56</span></strong> and <strong><span class="hashtag">#57</span></strong> respectively. From the <strong>response</strong> of the API, you can see when the runs were triggered (<strong>created_at</strong>).</p>
<ul>
<li>
<p>If time interval between <strong><span class="hashtag">#56</span></strong> and <strong><span class="hashtag">#57</span></strong> is more than but close to 24 hours, the scheduled workflow should have been triggered but just has a relatively long delay time.</p>
</li>
<li>
<p>If the time interval is close to (event more than) 48 hours, the scheduled workflow indeed was not triggered at that day. This may be caused by some issues on GitHub.</p>
</li>
</ul>

**Comment by kingthorin** [comment]
<p>Thanks <a class="mention" href="/brightran">@brightran</a></p>
<p>I’m okay with delays, I think I’ve seen that regularly, but it totally being skipped caught me off guard.</p>
<p>Does anyone know if there are any plans to increase assurance that things run, or to implement some sort of notification(s) if they don’t?</p>
<p>Edit: Just in-case anyone come across this later. The “run id”  for the API mentioned above isn’t just 56 or 57 it’s the one you see in the URL when viewing the run in the GitHub web UI.</p>
<p>Edit2: It seems 56 was created on the 17th (<code>"created_at": "2020-09-17T00:32:30Z",</code>), while 57 was created on the 19th (<code>"created_at": "2020-09-19T00:32:40Z",</code>). So there really does appear to have been no attempt to run it on the 18th :frowning:</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/kingthorin">@kingthorin</a> ,</p>
<blockquote>
<p>It seems 56 was created on the 17th ( <code>"created_at": "2020-09-17T00:32:30Z",</code> ), while 57 was created on the 19th ( <code>"created_at": "2020-09-19T00:32:40Z",</code> ).</p>
</blockquote>
<p>Looks like, the scheduled workflow indeed was not triggered on 18th.<br>
I noticed a GitHub incident reported on 17th (see <a href="https://www.githubstatus.com/incidents/ls7h2sdydbhl" rel="noopener nofollow ugc">here</a>). Maybe the issue is related to this incident.</p>

**Comment by AnandChowdhary** [comment]
<p>I wrote a quick blog post with some additional information about this issue: <a href="https://upptime.js.org/blog/2021/01/22/github-actions-schedule-not-working" rel="noopener nofollow ugc">https://upptime.js.org/blog/2021/01/22/github-actions-schedule-not-working</a>.</p>
